components/suggestion/suggestion_box/suggestion_box.jsx

```diff
@@ -638,9 +638,10 @@ export default class SuggestionBox extends React.PureComponent {
         const terms = suggestions.terms;
         const items = suggestions.items;
         let selection = this.state.selection;
-        if (terms.length > 0) {
+        const selectionIndex = terms.indexOf(selection);
```

Comment:
@hmhealey @Willyfrog This will resolve the selection problem but I am not sure if this will be enough, I was thinking of maintaining a state of the selected index as well. And check if the index of the selected term matches with the index in the new terms array received.

Reply:
Currently, we maintain a state of the selected term, which is getting reset as soon as we receive new items this change will fix that problem.